### ISSUE TITLE
Celery worker raises ValueError(gid)

### ISSUE THREAD

**Issue by blubber** [celery/celery#1954]
Running a chord generates the following exception:

```
[2014-04-04 11:49:55,635: ERROR/Worker-3] Chord callback '7c7f16d5-2d39-4721-9c39-62ed9495b770' raised: ValueError('7c7f16d5-2d39-4721-9c39-62ed9495b770',)
Traceback (most recent call last):
  File "./venvs/backend-33/lib/python3.3/site-packages/celery/backends/base.py", line 540, in on_chord_part_return
    raise ValueError(gid)
ValueError: 7c7f16d5-2d39-4721-9c39-62ed9495b770
```

```
$ celery report
software -> celery:3.1.10 (Cipater) kombu:3.0.14 py:3.3.2
            billiard:3.3.0.16 redis:2.9.1
platform -> system:Darwin arch:64bit imp:CPython
loader   -> celery.loaders.default.Loader
settings -> transport:redis results:redis://localhost:6379/0

CELERY_CREATE_MISSING_QUEUES: True
CELERY_DEFAULT_QUEUE: 'default'
CELERY_SEND_TASK_ERROR_EMAILS: True
BROKER_URL: 'redis://localhost:6379/0'
ADMINS: [redacted]
CELERY_DEFAULT_ROUTING_KEY: '********'
CELERY_ENABLE_UTC: True
CELERY_DEFAULT_EXCHANGE_TYPE: 'direct'
CELERY_RESULT_BACKEND: 'redis://localhost:6379/0'
SERVER_EMAIL: 'redacted'
CELERY_IMPORTS: ['core.tasks']

master::backend-33 wega: backend >
```

This is on python3.3 newest versions of Celery and Kombu

**Comment by ask** [comment]
Closed du to inactivity

**Comment by RumanticK** [comment]
will it be fixed in any further relises? Or we are stuck to 3.1.9?

**Comment by actionless** [comment]
I suspect what it can be related to https://github.com/celery/celery/issues/2573

In my app i was using celery `3.1.18` and such flow for the task:
`chord(group(task1, task2), task_fin)`
and it was working.

When i added two chained subtasks:
`chord(group(task1, task2,task3|task3_callback), task_fin)`
i started to experience the problem described in this issue.

So i tried to downgrade to celery `3.1.9`, with it i had behavior like in #2573 in my app

I recreated the same task workflow but without passing data between the tasks. In `3.1.18` it was not raising this (#1954) issue no more.

So, in conclusion, with the same task workflow, i have either this, either #2573 issue on depend if some data is being passed between the tasks. But without adding chained subtasks to the flow i am not experiencing neither of problems.

**Comment by ziziermao** [comment]
I am still seeing this at version 3.1.19

**Comment by ask** [comment]
@ziziermao Using ?new_join=1?

**Comment by jegesh** [comment]
I'm getting this with celery 3.1.24, using memcache result backend, with a very straightforward chord, i.e. `chord([mytask1.s(x) for x in list], mytask2.s()).apply_sync()`.  Solved it by changing to Redis with the ?new_join=1 param.
